.github/workflows/build.yml

```diff
@@ -142,7 +144,9 @@ jobs:
 
   build-dockers:
     runs-on: ubuntu-latest
-    needs: [build, build-arm]
+    needs: [build]
+    # run if `build` succeeded
+    if: ${{ always() && !cancelled() && needs.build.result == 'success' }}
```

Comment:
What does this mean? Isn't `always() && X` basically the same as `X`? Also, isn't this expression mirroring the default semantics?

Reply:
It's from here: https://stackoverflow.com/a/70959885

It probably has some effect, not sure what exactly though.